### ISSUE TITLE
Trailing space line breaks are removed from doc comments

### ISSUE THREAD

**Issue by BartMassey** [rust-lang/rustfmt#5639]
Consider this:

```
/*!
This  
has  
breaks.
*/
```

Note the two trailing spaces at each of lines 2 and 3. These are line breaks according to the Markdown specification. Unfortunately, `rustfmt` treats them as an accident and removes them.

The alternative markup here is `<br/>`, which is HTML-specific and harder to read, but won't be eaten by `rustfmt`.

I would recommend allowing exactly two spaces at the end of a line in doc comments, under the assumption that this is intended.

**Comment by BartMassey** [comment]
(`<br/>` is also embedded HTML, which means that tools for translating Markdown into other formats won't typically deal with it well.)

**Comment by ytmimi** [comment]
@BartMassey thanks for reaching out.

I believe that we [handle this](https://github.com/rust-lang/rustfmt/blob/ee2bed96d60fd7e46b1fb868f6a8f27e3a8058d0/src/comment.rs#L878), but I think rustfmt expects that block comments will be written with a leading `*` on each line. Is this still an issue when you write the following?

```rust
/*!
 * This  
 * has  
 * breaks.
 */
```

**Comment by ytmimi** [comment]
Just to be sure, no configuration options are needed to trigger this issue for you?

**Comment by ytmimi** [comment]
Just to be sure, no configuration options are needed to trigger this issue for you?

**Comment by BartMassey** [comment]
You are correct: if lines begin with leading `*` it looks like `rustfmt` leaves them alone. Interesting!

However, one major reason to use multiline doc comments is to avoid have formatting on the left; it is still important not to make semantic changes to the Markdown in this case.

No special configuration is needed: my repro example above is being run in a stock `rustfmt` environment. Please let me know if you can't reproduce, and I will try to investigate.
